MicroLite/SqlUtility.cs

```diff
@@ -119,16 +120,17 @@ public static string RenumberParameters(string sql, int totalArgumentCount)
             var argsAdded = 0;
             var parameterPrefix = parameterNames[0].Substring(0, parameterNames[0].IndexOfAny(digits));
 
-            var predicateReWriter = new StringBuilder(sql);
+            var result = sql;
 
-            foreach (var parameterName in parameterNames.OrderByDescending(n => n))
+            foreach (var parameterName in parameterNames.OrderByDescending(n => n.Replace(parameterPrefix, string.Empty).PadLeft(9, '0')))
             {
                 var newParameterName = parameterPrefix + (totalArgumentCount - ++argsAdded).ToString(CultureInfo.InvariantCulture);
 
-                predicateReWriter.Replace(parameterName, newParameterName);
+                var regex = new Regex(string.Format(CultureInfo.InvariantCulture, "({0})(?=${{1}}|\\D{{1}})", parameterName));
```

Comment:
Creating a new regex per parameter is expensive, this would run a lot slower with much more overhead than before... can we find a better way to do this?